frontend/components/ShareButton.tsx

```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+import { Button, useClipboard, useToast } from '@chakra-ui/react';
+import { FaShare } from 'react-icons/fa';
+
+const ClientShareButton = ({ pageAccount }: { pageAccount: string | null }) => {
+  if (!pageAccount) {
+    return null;
+  }
+  const currentUrl = `${window.location.host}/grave/${pageAccount}`;
+  const { hasCopied, onCopy } = useClipboard(currentUrl);
+  const toast = useToast();
+
+  const handleShare = () => {
+    onCopy();
+    toast({
+      title: 'Link copied to clipboard.',
+      status: 'success',
+      duration: 3000,
+      isClosable: true,
+    });
+  };
+
+  return (
+    <Button
+      colorScheme="blue"
+      onClick={handleShare}
+      size={'sm'}
+      rightIcon={<FaShare />}
+    >
+      {hasCopied ? 'Copied' : 'Share'}
```

Comment:
Should the default text be "copy" instead of share?

It'd probably be annoying to override the user's clipboard if it's not explicit

Reply:
will modify share button to be better. agree w/ you point

Reply:
<img width="559" alt="Screenshot 2024-01-29 at 1 13 50 PM" src="https://github.com/yearone-io/universal-grave/assets/128236767/15b7041d-8cba-49c5-80e6-135a79a2bc12">
will have dropdown instead